lib/common/models/base.rb

```diff
@@ -5,12 +5,11 @@
 module Common
   # This is a base serialization class
   class Base
-    include Comparable
-    include ActiveModel::Serialization
     extend ActiveModel::Naming
+    include ActiveModel::Serialization
+    include Comparable
     include Virtus.model(nullify_blank: true)
 
-    attr_reader :attributes
     attr_accessor :metadata, :errors_hash
     alias to_h attributes
     alias to_hash attributes
```

Comment:
I think you can remove these lines as well, they are defined in Virtus
https://github.com/solnic/virtus/blob/master/lib/virtus/instance_methods.rb#L45